### DISCUSSION TITLE
Expose more types

### DISCUSSION THREAD

**Discussion by MoSattler** [payloadcms/payload#2090] [Feature Requests & Ideas]
https://github.com/payloadcms/payload/pull/2061 was already super helpful, thank you for that! 

However, I think that if more types were exported, it would be even more useful.

To give you an example, typing access:
```ts
import type { Access } from "payload/config";

const isAdminOrCreatedBy: Access = ({ req: { user } }) => {/* code */}
```
is not great, since it does a lot of guessing. E.g. `user` is `any`. 

What I would love to be able to do, is this:

```ts
import type { AccessArgs } from "payload/config";
import type { User } from "payload/generated-types";

const isAdminOrCreatedBy = <T>({ req: { user } }: AccessArgs<T, User>) => {/* code */}
```

here I profit from the generic typing for `data` and have already a clearly defined `user`. 

The type `AccessArgs` already exists, and only needs to be exported to the consumer.

**Comment by MoSattler** [comment]
Another scenario is when defining access methods inline.

```ts
{
    access: {
      read: ({ req: { user }, data }: AccessArgs<MyCollection, User>) => {
        /* code */
      },
  },
}
```

without `AccessArgs` I cannot easily type the function args.

**Comment by JessRynkar** [comment]
Should also add more generic collection types, see related conversation [here](https://discord.com/channels/967097582721572934/1075948267789090896).
